### DISCUSSION TITLE
HAN red but updates seems to be fine

### DISCUSSION THREAD

**Discussion by rogerthn2019** [UtilitechAS/amsreader-firmware#626] [General]
See below and please advice
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/57598997/0f176195-d580-4877-9c33-bc109f0764fb)
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/57598997/9928a9ab-2a6f-4bca-9168-7bf64b9f5a27)

**Comment by ArnieO** [comment]
What hardware you do use: Is it a Pow-K/Pow-K+, or is it your own custom hardware on an ESP32?

The HAN indicator turns red and stays red for a while if payloads are rejected or not received as expected. If payloads are often received with errors, you will see this. 

To investigate further, please run a [Telnet debug](https://amsleser.no/blog/post/24-telnet-debug) and post a screenshot of the output. 
Please also provide screenshots of your info- and config pages.

**Comment by rogerthn2019** [comment]
Pow-K+ as sent by yourself  2023-03-30 "Sendes i dag, sporingsnummer UK007109149NO"
Seems to be steady red and updating.
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/57598997/951cace2-8643-4611-ac58-c6345b67729b)

Enclosing telnet output in telnet.txt.
[telnet.txt](https://github.com/UtilitechAS/amsreader-firmware/files/12478945/telnet.txt)
When telnet was disabled it did reboot and HAN is green again.

**Reply by gskjold** [reply]
> When telnet was disabled it did reboot and HAN is green again

That was some really useful information (combined with your high uptime) that led me right to the issue, thank you! 

Technical explanation: The frameworks 32bit "millis()" function rolls over at 49.7 days. This is currently used in the data container, but will be fixed in future release by using 64bit instead

**Reply by rogerthn2019** [reply]
OK!
I'm not sure that I'll be here 2038 but there might be some 32-bit system still running ;-)

**Reply by ArnieO** [reply]
🤣
